en/components/ripple.md

```diff
@@ -47,162 +40,159 @@ export class AppModule {}
+
+The `igx-palette` function generates a color palette based on the primary and secondary colors that are passed:
 
 ```scss
-$my-custom-schema: extend($dark-schema, (
-    igx-ripple: $custom-ripple-schema
-));
+$green-color: #217346;
+$black-color: #151515;
 
-$custom-theme: igx-ripple-theme(
-    $palette: $custom-palette,
-    $schema: $my-custom-schema
+$custom-ripple-palette: igx-palette(
+    $primary: $black-color,
+    $secondary: $green-color
 );
-```
+```   
+
+After the custom palette has been generated, the `igx-color` function can be used to obtain different varieties of the primary and the secondary colors.   
 
-#### Applying the custom theme
-The easiest way to apply your theme is with a `sass` `@include` statement in the global styles file:
 ```scss
-@include igx-ripple($custom-theme);
+$custom-ripple-theme: igx-ripple-theme(
+    $color: igx-color($custom-ripple-palette, "secondary", 500)
+);
 ```
 
-#### Scoped component theme
+>[!NOTE]
+>The `igx-color` and `igx-palette` are powerful functions for generating and retrieving colors. Please refer to the [`Palettes`](themes/palette.md) topic for detailed guidance on how to use them.
 
-In order for the custom theme to affect only specific component, you can move all of the styles you just defined from the global styles file to the custom component's style file (including the import of the `index` file).
+#### Using schemas
 
-This way, due to Angular's [ViewEncapsulation](https://angular.io/api/core/Component#encapsulation), your styles will be applied only to your custom component.
+You can go even further and build flexible structure that has all the benefits of a [**schema**](../themes/schemas.md), which is the recipe of a theme. 
 
- >[!NOTE]
- >If the component is using an [`Emulated`](../themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to penetrate this encapsulation using `::ng-deep` in order to style the grid.
- >[!NOTE]
- >Wrap the statement inside of a `:host` selector to prevent your styles from affecting elements *outside of* our component:
+Extend one of the two predefined schemas, that are provided for every component. In our case, we would use [`light-ripple`]({environment:sassApiUrl}/index.html#variable-_light-ripple).   
 
 ```scss
-:host {
-    ::ng-deep {
-        @include igx-ripple($custom-theme);
-    }
-}
-```    
+//  Extending the ripple light schema
+$custom-ripple-schema: extend($_light-ripple, (
+    color: (igx-color("secondary", 500))
+));
+   
+// Defining ripple with the custom light schema
+$custom-ripple-theme: igx-ripple-theme(
+    $palette: $custom-ripple-palette,
+    $schema: $custom-ripple-schema
+);
+```
+
+Don't forget to include the themes in the same way as it was demonstrated above. 
 
 >[!NOTE]
   > A color that is set using the `igxRiple` directive, would take precedence from the one, set within a custom theme.  
```

Comment:
This is not needed. Palettes are explained in their own section. Demonstrating how schemas work without context adds more confusion.